### ISSUE TITLE
Pose outputs for custom dataset

### ISSUE THREAD

**Issue by ArthurMoreau** [nianticlabs/monodepth2#203]
Hi !
I'm training a model with my custom dataset, starting from your kitti weights.
Depth training & inference are very good but I also would like to generate car trajectory given a video input.
For a scenario of loop closure (the car come back to its initial position), pose network with kitti pretrained weights is quite ok (no training with my data).

![image](https://user-images.githubusercontent.com/53355442/84895612-d8fef700-b0a2-11ea-8ae2-98d4fadbbc8d.png)

But i when I fine tune on my data, depth prediction improves a lot while pose prediction predicts too small axis angles

![image](https://user-images.githubusercontent.com/53355442/84895730-106da380-b0a3-11ea-9a9a-b7e3e9283c3d.png)

The weird thing is that if I multiply these angles from pose network by some scale factor, I can get very good results

![image](https://user-images.githubusercontent.com/53355442/84895896-5460a880-b0a3-11ea-856e-e52a210aa9d1.png)

In other words, translations are ok and rotations are too small (but correct up to a multiplier factor).

Of course, I changed the calibration settings to adapt to my camera.

Any of you already faced this problem or has an idea how to solve it ?

**Comment by mrharicot** [comment]
Hi,
That's a very interesting problem!

How did you pick that scale factor to "fix" the rotations?
Did you try training on your data from scratch?

Monodepth2 has a bad habit of underestimating the rotations.

**Comment by ArthurMoreau** [comment]
I found the scale "manually" with trials. The value is around 4, so I thought that maybe it was relative to the scale factor to convert relative depth to absolute but there is no reason for that. Plus kitti model returns quite correct angles without any scaling.

Training from scratch give the same pose results with less accurate depth.

**Comment by mdfirman** [comment]
This is a good observation, and I think that Clément is correct here – Monodepth2 is good at depth, but lazy at pose (esp. rotations). 

We had a hypothesis that training with 'average' reprojection loss might improve odometry results, but I don't think we ever did that experiment. 

There are some newer papers that really focus on getting good poses with the depths (e.g. D3VO?), which are more likely to do good things with your data.

Thanks

**Comment by YJonmo** [comment]
> This is a good observation, and I think that Clément is correct here – Monodepth2 is good at depth, but lazy at pose (esp. rotations).
> 
> We had a hypothesis that training with 'average' reprojection loss might improve odometry results, but I don't think we ever did that experiment.
> 
> There are some newer papers that really focus on getting good poses with the depths (e.g. D3VO?), which are more likely to do good things with your data.
> 
> Thanks

Could you mention some of those papers which do a better job in pose estimation? 

Thanks
